### PR TITLE
Change to toml for SourceGit-CentralGit sync, search for image-mappings explicitly, add additional data validations

### PR DESCRIPTION
**Pull request by avinashtrivedi11**
- using toml file for mapping paths
- added support for image mappings(both source and central)
- Image paths get updated to absolute path in both pipelines even if relative path is mentioned
- central pipeline to only trigger on a change from DKP using `workflow:rules`
- Source Checks: filesize, spaces, path existence,
- Central checks: filesize

Tested on `dt-iva-cicd-playground` and `demo source` repos.

### CODE REVIEW

File: knowledge-portal/central/README.md [hunk 1]
```diff
@@ -1,48 +1,60 @@
-# Setting up pipeline in central repository
+# Knowledge Portal Central Pipeline
```
Comment (rsnk96) [inline knowledge-portal/central/README.md:1]: *Central Git

File: knowledge-portal/central/initiate-push-to-source.sh [hunk 2]
```diff
@@ -2,21 +2,12 @@
...
-do
-    commit_message=$(git log -1 --format=%B);
-    if [[ ${commit_message} == *"docs:"* ]]
-    then
-        git reset --soft HEAD~1
-        ((commits_to_check++))
-    else
-        break
-    fi
-done
```
Comment (rsnk96) [inline knowledge-portal/central/initiate-push-to-source.sh:17]: Are we removing the system where we reset to the n-th last commit ?

File: knowledge-portal/central/README.md [hunk 3]
```diff
@@ -1,48 +1,60 @@
...
-## Setting up Access Token in central repository
+### Central Pipeline Checks
 
-1. Go to `Settings >> Access Tokens`.
-2. Set `Token name = PUSH_TOKEN`. This is important and you can't choose any other name without modifying the source pipeline as well.
-3. Select the role `Maintainer` and `write_repository` as the scope of token.
-4. `Create project access token` will create a unique token for you that can then be shared with `maintainers` setting up individual `source` pipelines.
+Before the job in pipeline pushes files back to `source`, it checks multiple things based on your mapping file. Knowing about these checks will help you use the pipeline easily and make any necessary changes before any change on wikijs become available at `source` repository.
 
+- The jobs will only run when a commit happens on `main` branch with a message starting with `docs:`, This ensures that the jobs will only run when a commit has been made from wikijs portal.
```
Comment (rsnk96) [inline knowledge-portal/central/README.md:25]: main branch of CentralGit, I take it...? Can we specify it explicitly?

File: knowledge-portal/central/README.md [hunk 4]
```diff
@@ -1,48 +1,60 @@
...
+- Ensure that all image URL paths in Markdown files are absolute and begin with `/`.
```
Comment (rsnk96) [inline knowledge-portal/central/README.md:26]: I presume this is for local image files, not ones on the web (Ex: `http` or `https`). If yes, can we specify that as a subpoint?

File: knowledge-portal/central/README.md [hunk 5]
```diff
@@ -1,48 +1,60 @@
...
+- Ensure all image URL paths in a Markdown file are in downstream of that Markdown file.
```
Comment (rsnk96) [inline knowledge-portal/central/README.md:27]: > "...are in downstream of that Markdown file"
We can add something along the lines of "...are downstream in the folder structure wrt the Markdown file", and then add an correct and wrong example as a subpoint.

File: knowledge-portal/source/README.md [hunk 6]
```diff
@@ -1,61 +1,67 @@
-# Usage
+# Knowlege portal Source Pipeline
```
Comment (rsnk96) [inline knowledge-portal/source/README.md:1]: *Source Git :)

File: knowledge-portal/source/README.md [hunk 7]
```diff
@@ -1,61 +1,67 @@
...
+
+include:
+  - remote: 'https://github.com/detecttechnologies/Gitlab-CI-CD-Templates/raw/main/knowledge-portal/source/.gitlab-ci.yml'
+
+variables:
+  GIT_STRATEGY: clone
 
 ```
-## Configure CI/CD variables (applied on overall gitlab instance)
+### Using the toml file to connect to Knowledge Portal
```
Comment (rsnk96) [inline knowledge-portal/source/README.md:20]: *to connect your source git repository to the knowledge portal

File: knowledge-portal/source/README.md [hunk 8]
```diff
@@ -1,61 +1,67 @@
...
 
-1. Go to `Settings >> CI/CD >> Variables`
-2. Click on `Add variable`. 
-3. Give the name `CENTRAL_GIT_PUSH_URL` in `Key` field and `Value` field will be provided by your maintainers.
-    * For maintainers: The `CENTRAL_GIT_PUSH_URL` will be of the form: `https://oauth2:$PUSH_TOKEN@gitlab.com/<org-name>/<path-to-central-git-repo>.git`
-4. Click on `Add variable` button to confirm changes. 
+Here is an example of a `docs-manifest.toml` file that you can use to create your own and connect your pipeline to knowledge portal:
```
Comment (rsnk96) [inline knowledge-portal/source/README.md:22]: *connect your ~pipeline~ repository

File: knowledge-portal/source/README.md [hunk 9]
```diff
@@ -1,61 +1,67 @@
...
 
 
-> The steps of this section are being performed on the overall gitlab instance, so that it need not be replicated for every repo.
+```toml
+# "includes" section maps source paths to paths at Knowledge Portal
```
Comment (rsnk96) [inline knowledge-portal/source/README.md:26]: Can perhaps reword to make it easier to read. E.g.

"includes" section maps paths inside your source Git repository to paths that the documents should take within the Knowledge Portal

File: knowledge-portal/central/README.md (hunk 5 above)
Comment (avinashtrivedi11) [reply knowledge-portal/central/README.md:27]: this wouldn't apply, sorry this was from old doc. Guess, i forgot to modify in Central pipeline. We are allowing relative paths and upstream directories as well now.

File: knowledge-portal/central/initiate-push-to-source.sh (hunk 2 above)
Comment (avinashtrivedi11) [reply knowledge-portal/central/initiate-push-to-source.sh:17]: Yeah, the pipeline will only take the latest commit.

File: knowledge-portal/source/README.md (hunk 6 above)
Comment (avinashtrivedi11) [reply knowledge-portal/source/README.md:1]: done

File: knowledge-portal/source/README.md (hunk 7 above)
Comment (avinashtrivedi11) [reply knowledge-portal/source/README.md:20]: done, instead used "Connect your Source Git repository to the Knowledge Portal"

File: knowledge-portal/source/README.md (hunk 8 above)
Comment (avinashtrivedi11) [reply knowledge-portal/source/README.md:22]: done

File: knowledge-portal/source/README.md (hunk 9 above)
Comment (avinashtrivedi11) [reply knowledge-portal/source/README.md:26]: done

File: knowledge-portal/central/README.md (hunk 4 above)
Comment (avinashtrivedi11) [reply knowledge-portal/central/README.md:26]: done

File: knowledge-portal/central/README.md (hunk 3 above)
Comment (avinashtrivedi11) [reply knowledge-portal/central/README.md:25]: done

File: knowledge-portal/central/README.md (hunk 1 above)
Comment (avinashtrivedi11) [reply knowledge-portal/central/README.md:1]: done

**Pull request merged**